### ISSUE TITLE
OpenResty / Docker integration

### ISSUE THREAD

**Issue by hisk** [pkulchenko/ZeroBraneStudio#998]
I'm trying to integrate ZeroBrane debugging with OpenResty running in a Docker container but I'm having trouble getting it set up. I've read most of the docs and followed [this guide](http://notebook.kulchenko.com/zerobrane/debugging-openresty-nginx-lua-scripts-with-zerobrane-studio), but the breakpoints are simply not triggered and there are no errors whatsoever.

**My setup is as follows:**

* ZeroBrane is located in Docker's build context and it's mapped as a volume inside the container. Not sure if this is the best approach, but I thought having it there would help.
* I've mapped the `lua_package_path` and `lua_package_cpath` appropriately and they can be accessed from the container.
* I've used the host ip (my machine's) in `require('mobdebug').start('<host-ip>')`. Tried other options as well, but no luck.

I was wondering if there is anyone who managed to get a similar setup up and running, or if you have any suggestions that might help. Thank you!

**Comment by pkulchenko** [comment]
@hisk, see the discussion in #967, as I suspect the reason may be the same. If the debugging is actually starting (so the connection from the application gets to the IDE), then the most likely reason for breakpoints not being hit is a path mismatch (you may find more details on how to check this and also other reasons in this section: https://studio.zerobrane.com/doc-faq#why-breakpoints-are-not-triggered).

**Comment by moteus** [comment]
For this case i have [simple patch](https://github.com/moteus/ZeroBraneStudio/blob/moteus_work/src/editor/debugger.lua#L315) to change file path
On my work I also run server in the docker and run ZBS on the host system

**Comment by pkulchenko** [comment]
@moteus, good point. I've seen the patch, but I'm not convinced it's needed; or at least would like to figure out why it's needed, as the IDE is supposed to do the path re-mapping to handle cases like this.

Do you know why [debugger:mapRemotePath](https://github.com/pkulchenko/ZeroBraneStudio/blob/master/src/editor/debugger.lua#L547) is not triggered?

**Comment by moteus** [comment]
> Do you know why debugger:mapRemotePath is not triggered?

Not sure but i think it works only with paths with some common prefixes
In my use case i have 2 sets of Lua source code - server's core library and project code.
This is 2 independent repo and have 2 different paths in the docker. An i need to match them to 2 different paths in my home dir

**Comment by moteus** [comment]
Does ZBS has such concept as projects config.
I think i can write plugin to handlin on project load, but may be there exists one?

**Comment by pkulchenko** [comment]
> Does ZBS has such concept as projects config. I think i can write plugin to handlin on project load, but may be there exists one?

@moteus, yes, [projectsettings](https://github.com/pkulchenko/ZeroBranePackage/blob/master/projectsettings.lua)

> In my use case i have 2 sets of Lua source code - server's core library and project code. This is 2 independent repo and have 2 different paths in the docker. An i need to match them to 2 different paths in my home dir

This should still work as long as the core library and project code have some common path inside the container and outside of it. Do you have a specific example of how the core library and project code paths look inside the Docket and outside?

**Comment by moteus** [comment]
> Do you have a specific example of how the core library and project code paths look inside the Docket and outside?

This is what I have [right now](https://github.com/moteus/zbs-package/blob/master/zbstudio/user.lua#L186). All thouse files in `/usr/share` installed via deb packages in the docker. And it is 2 different independed projects and it is not `work` project.

**Comment by rulatir** [comment]
@moteus, does your last comment mean that you resolved the issue, or did you merely supply the information @pkulchenko asked for? Did you resolve the issue in the meantime, or are you still waiting for @pkulchenko to reply?

**Comment by moteus** [comment]
> @moteus, does your last comment mean that you resolved the issue, or did you merely supply the information @pkulchenko asked for? Did you resolve the issue in the meantime, or are you still waiting for @pkulchenko to reply?

My patch allows me to run a server (not nginx, but also some HTTP server) in the docker container and do debug.
If you want you can try my patch and check out either it solve your problem. I think it is will be very helpful to make a decision about how useful it is.
